tests/benchmark/utils/s2n_server_recv_benchmark.cc

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tests/benchmark/utils/s2n_server_recv_benchmark.h"
+#include "tests/benchmark/utils/shared_info.h"
+#include "utils/s2n_safety.h"
+#include <unistd.h>
+#include <stdio.h>
+#include <vector>
+#include <stdlib.h>
+#include <poll.h>
+
+extern "C" {
+#include "bin/common.h"
+#include "tls/s2n_connection.h"
+}
+
+int recv_data(struct s2n_connection *conn, int sockfd2, benchmark::State& state, bool warmup) {
+    struct pollfd readers[1];
+    readers[0].fd = sockfd2;
+    readers[0].events = POLLIN;
+    errno = 0;
+    int poll_output = 0;
+    s2n_blocked_status blocked;
+    uint8_t* recieved;
+    recieved = (uint8_t*)malloc(state.range(0) + 2);
+
+    while ((poll_output = poll(readers, 1, -1)) > 0) {
```

Comment:
What's going on here. I'm not familiar with s2n send/recv. Why are polling, and where does this data go?

Reply:
poll() checks to see if the file descriptor is ready. I didn't want to try to s2n_recv any data if there was not data coming in because the benchmark times each call to s2n_recv.